fluentd/datadog_checks/fluentd/fluentd.py

```diff
@@ -90,19 +95,32 @@ def check(self, instance):
 
             self._collect_metadata()
         except Exception as e:
-            msg = "No stats could be retrieved from %s : %s" % (url, str(e))
+            msg = "No stats could be retrieved from %s : %s" % (self.url, str(e))
             self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, tags=service_check_tags, message=msg)
             raise
         else:
             self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.OK, tags=service_check_tags)
 
     def _collect_metadata(self):
-        if self.is_metadata_collection_enabled():
-            raw_version = self._get_raw_version()
-            if raw_version:
-                self.set_metadata('version', raw_version)
+        raw_version = None
+        if not self.is_metadata_collection_enabled():
+            return
+
+        try:
+            r = self.http.get(self.config_url)
+            r.raise_for_status()
+            config = r.json()
+            raw_version = config.get('version')
+        except Exception as e:
+            self.log.debug("No config could be retrieved from %s: %s", self.config_url, e)
```

Comment:
Nit: we can simplify the logic a bit by applying the fallback here, dropping the need for `raw_version = None` initially. Maybe we can also embed the minimal fluentd version for background:

```suggestion
            self.log.debug("No config could be retrieved from %s: %s", self.config_url, e)
            # Fall back to command line as this may be an older version of Fluentd (< 1.8.0)
            raw_version = self._get_version_from_command_line()
```

Reply:
`raw_version = config.get('version')`
Will return `None` (no exception) if the request to the config endpoint is succesfull but is an older version of fluentd